drivers/tee/optee/core.c

```diff
@@ -487,12 +496,13 @@ optee_config_shm_memremap(optee_invoke_fn *invoke_fn, void **memremaped_shm,
 		vaddr += sz;
 		paddr += sz;
 		size -= sz;
+
+		rc = tee_shm_pool_mgr_alloc_res_mem(vaddr, paddr, size, PAGE_SHIFT);
+		if (IS_ERR(rc))
+			goto err_free_priv_mgr;
+		dmabuf_mgr = rc;
 	}
```

Comment:
It would be more friendly to add an `else` here with an error message and appropriate error code.